src/UnitGenerator/SourceGenerator.cs

```diff
@@ -119,35 +133,81 @@ internal class UnitOfAttribute : Attribute
     {
         public Type Type { get; }
         public UnitGenerateOptions Options { get; }
-        public string Format { get; }
+        public UnitArithmeticOperators ArithmeticOperators { get; set; }
+        public string ToStringFormat { get; set; }
 
-        public UnitOfAttribute(Type type, UnitGenerateOptions options = UnitGenerateOptions.None, string toStringFormat = null)
+        public UnitOfAttribute(Type type, UnitGenerateOptions options = UnitGenerateOptions.None)
         {
             this.Type = type;
             this.Options = options;
-            this.Format = toStringFormat;
         }
     }
-
+    
     [Flags]
     internal enum UnitGenerateOptions
     {
         None = 0,
         ImplicitOperator = 1,
-        ParseMethod = 2,
-        MinMaxMethod = 4,
-        ArithmeticOperator = 8,
-        ValueArithmeticOperator = 16,
-        Comparable = 32,
-        Validate = 64,
-        JsonConverter = 128,
-        MessagePackFormatter = 256,
-        DapperTypeHandler = 512,
-        EntityFrameworkValueConverter = 1024,
-        WithoutComparisonOperator = 2048,
-        JsonConverterDictionaryKeySupport = 4096
+        ParseMethod = 1 << 1,
+        MinMaxMethod = 1 << 2,
+        ArithmeticOperator = 1 << 3,
+        ValueArithmeticOperator = 1 << 4,
+        Comparable = 1 << 5,
+        Validate = 1 << 6,
+        JsonConverter = 1 << 7,
+        MessagePackFormatter = 1 << 8,
+        DapperTypeHandler = 1 << 9,
+        EntityFrameworkValueConverter = 1 << 10,
+        WithoutComparisonOperator = 1 << 11,
+        JsonConverterDictionaryKeySupport = 1 << 12,
+    }
+
+    [Flags]
+    internal enum UnitArithmeticOperators
+    {
+        Number = 0,
+        Addition = 1,
+        Subtraction = 1 << 1,
+        Multiply = 1 << 2,
+        Division = 1 << 3,
+        Increment = 1 << 4,
+        Decrement = 1 << 5,
     }
+    
+#if NET7_0_OR_GREATER
+   internal static class AsNumber<T> where T : INumber<T>
+   {
+        public static T One => T.One;
```

Comment:
Primitive types such as Int32 explicitly implement the `INumber<>` member, so cannot calls such as `int.One`. So I have created a proxy like this, but is there a better way to do it... ?